### ISSUE TITLE
Issue / Bug with Boolean CommandParameter from Xaml in MvxCommand

### ISSUE THREAD

**Issue by crazyfx1** [MvvmCross/MvvmCross#836]
Hi,

To expand our application from Windows Phone and Windows Store to Xamarin, we are currently migrating our Project from Galasoft MvvmLight to MvvmCross. 

Unfortunately, there seems to be a problem with the MvxCommand using a bool Parameter. The problem occurs in both Windows Phone and Windows Store projects.

When I create a button in Xaml using a Binding to a MvxCommand and a CommandParameter which is a Boolean-Text, an Exception is thrown.
Xaml:

```
    <Button Command="{Binding Path=TestCommand}" CommandParameter="True" />
```

Command (the actual Execute / CanExecute-Action does not matter on this issue):

```
    public MvxCommand<bool> TestCommand;
```

Exception:

```
    Specified cast is not valid.
```

StackTrace:

```
   at Cirrious.MvvmCross.ViewModels.MvxCommand`1.CanExecute(Object parameter)
   at System.Windows.Controls.Primitives.ButtonBase.UpdateCanExecute()
   at System.Windows.Controls.Primitives.ButtonBase.OnCommandChanged(DependencyPropertyChangedEventArgs e)
   at System.Windows.Controls.Primitives.ButtonBase.OnCommandPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
   at System.Windows.DependencyObject.RaisePropertyChangeNotifications(DependencyProperty dp, Object oldValue, Object newValue)
   at System.Windows.DependencyObject.UpdateEffectiveValue(DependencyProperty property, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, ValueOperation operation)
   at System.Windows.DependencyObject.RefreshExpression(DependencyProperty dp)
   at System.Windows.Data.BindingExpression.SendDataToTarget()
   at System.Windows.Data.BindingExpression.SourceAcquired()
   at System.Windows.Data.BindingExpression.System.Windows.IDataContextChangedListener.OnDataContextChanged(Object sender, DataContextChangedEventArgs e)
   at System.Windows.Data.BindingExpression.DataContextChanged(Object sender, DataContextChangedEventArgs e)
   at System.Windows.FrameworkElement.OnDataContextChanged(DataContextChangedEventArgs e)
   at System.Windows.FrameworkElement.OnAncestorDataContextChanged(DataContextChangedEventArgs e)
   at System.Windows.FrameworkElement.NotifyDataContextChanged(DataContextChangedEventArgs e)
   at System.Windows.FrameworkElement.OnAncestorDataContextChanged(DataContextChangedEventArgs e)
   at System.Windows.FrameworkElement.NotifyDataContextChanged(DataContextChangedEventArgs e)
   at System.Windows.FrameworkElement.OnPropertyChanged(DependencyProperty dp)
   at System.Windows.DependencyObject.RaisePropertyChangeNotifications(DependencyProperty dp, Object oldValue, Object newValue)
   at System.Windows.DependencyObject.UpdateEffectiveValue(DependencyProperty property, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, ValueOperation operation)
   at System.Windows.DependencyObject.SetValueInternal(DependencyProperty dp, Object value, Boolean allowReadOnlySet)
   at System.Windows.FrameworkElement.set_DataContext(Object value)
   at Cirrious.MvvmCross.WindowsPhone.Views.MvxPhonePage.set_ViewModel(IMvxViewModel value)
   at Cirrious.MvvmCross.Views.MvxViewExtensionMethods.OnViewCreate(IMvxView view, Func`1 viewModelLoader)
   at Cirrious.MvvmCross.WindowsPhone.Views.MvxPhoneExtensionMethods.OnViewCreate(IMvxPhoneView phoneView, Uri navigationUri, IMvxBundle savedStateBundle)
   at Cirrious.MvvmCross.WindowsPhone.Views.MvxPhonePage.OnNavigatedTo(NavigationEventArgs e)
   at xxx.xxx.WP8.Common.ServicePage.OnNavigatedTo(NavigationEventArgs e)
   at Microsoft.Phone.Controls.PhoneApplicationPage.InternalOnNavigatedTo(NavigationEventArgs e)
   at Microsoft.Phone.Controls.PhoneApplicationPage.Microsoft.Phone.Controls.IPhoneApplicationPage.InternalOnNavigatedToX(NavigationEventArgs e)
   at System.Windows.Navigation.NavigationService.RaiseNavigated(Object content, Uri uri, NavigationMode mode, Boolean isNavigationInitiator, IPhoneApplicationPage existingContentPage, IPhoneApplicationPage newContentPage)
   at System.Windows.Navigation.NavigationService.CompleteNavigation(DependencyObject content, NavigationMode mode)
   at System.Windows.Navigation.NavigationService.ContentLoader_BeginLoad_Callback(IAsyncResult result)
```

It seems like the CommandParameter is not parsed correctly. If I change the Parameter of the MvxCommand to string it works, and I get the "True"-String as Parameter. The parsing works in GalaSoft MvvmLight. 

Is this a bug? Is there any chance I can create a Bool already in Xaml so MvvmCross passes the Parameter correctly?

As a hint: MvvmLight casts the parameter to the target type before passing it into the Command.

Best regards
Martin

**Comment by nickrandolph** [comment]
@crazyfx1 is this still an issue in the latest release (6 or 6.1)?

**Comment by crazyfx1** [comment]
I fixed this with PR #842, which was merged in 3.5 or 4.0. So this issue should be closed.

**Comment by nickrandolph** [comment]
Thanks for that.
